src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -459,6 +459,30 @@ CHIP_ERROR DiscoveryImplPlatform::FinalizeServiceUpdate()
     return ChipDnssdFinalizeServiceUpdate();
 }
 
+void DiscoveryImplPlatform::RegisterResolverDelegate(ResolverDelegate * delegate)
+{
+    for (auto & resolverDelegate : mResolverDelegates)
+    {
+        if (resolverDelegate == nullptr || resolverDelegate == delegate)
+        {
+            resolverDelegate = delegate;
+            break;
+        }
+    }
```

Comment:
Maybe we can return an error or print some error message when we run out of empty slots?